tests/dataset/test_aoi.py

```diff
@@ -290,39 +308,63 @@ def test_raster_stats_not_stac(self) -> None:
-            aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], raster_bands_request=['red', 'green', 'blue'])
-            aoi.to_dict()
-            aoi.close_raster()
+        bands = ['red', 'green', 'blue']
+        row = next(iter(data))
+        aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], raster_bands_request=bands)
+        aoi.to_dict()
+        aoi.close_raster()
 
     def test_for_multiprocessing(self) -> None:
         """Tests multiprocessing on AOI instances"""
         extract_archive(src="tests/data/spacenet.zip")
         data = read_csv("tests/tiling/tiling_segmentation_multiclass_ci.csv")
         inputs = []
-        for row in data:
-            aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], for_multiprocessing=True)
-            inputs.append([aoi_read_raster, aoi])
-            assert aoi.raster_closed is True
-            assert aoi.raster is None
+        row = next(iter(data))
+        aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], for_multiprocessing=True)
+        inputs.append([aoi_read_raster, aoi])
+        assert aoi.raster_closed is True
+        assert aoi.raster is None
 
         with multiprocessing.get_context('spawn').Pool(None) as pool:
             aoi_meta = pool.map_async(map_wrapper, inputs).get()
         print(aoi_meta)
 
+    def test_name_raster(self) -> None:
+        """Tests naming of raster given multiple input type"""
+        inputs = [
+            ("tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-${dataset.bands}.tif",
+             ["R", "G", "B"]),
+            ("http://datacube-stage-data-public.s3.ca-central-1.amazonaws.com/store/imagery/optical/spacenet-samples/"
+             "SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-${dataset.bands}.tif", ["R", "G", "B"]),
+            ("tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03.json", ["red", "green", "blue"]),
+            ("tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB.tif", [1, 2, 3]),
+            ("tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB.tif", [1, 2, 3, 4, 5, 6, 7]),
+            ("tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB.tif", []),
+        ]
+        expected_list = [
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-bands_R-G-B.tif",
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-bands_R-G-B.tif",
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03_bands_red-green-blue.tif",
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB_bands_1-2-3.tif",
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB_7bands.tif",
+            "tests/data/spacenet/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-RGB.tif",
+        ]
```

Comment:
This give a good idea of the "raster_name" outputted according to different types of input.